### ISSUE TITLE
Python 3.7 pylint 2.0.0.dev2 AttributeError: module 'astroid.nodes' has no attribute 'CallFunc'

### ISSUE THREAD

**Issue by fsnidely** [PyCQA/pylint-django#173]
I'm trying to use pylint for Python 3.7, which isn't supported until pylint 2.0.0, so it is still in dev. However, I thought I would bring this here as pylint 2.0.0 fails when loading pylint_django, simply when checking the version (no file to be linted).

```
2062$ pylint --version
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.7.0 (default, Jun 28 2018, 02:32:19) 
[GCC 5.4.0 20160609]
```
Fails when loading the plugin pylint_django:
```
2061$ pylint --version --load-plugins pylint_django
Traceback (most recent call last):
  File "/home/ferd/.virtualenvs/python37/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint/__init__.py", line 18, in run_pylint
    Run(sys.argv[1:])
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint/lint.py", line 1293, in __init__
    linter.load_plugin_modules(self._plugins)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint/lint.py", line 502, in load_plugin_modules
    module = modutils.load_module_from_name(modname)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/astroid/modutils.py", line 187, in load_module_from_name
    return load_module_from_modpath(dotted_name.split('.'), path, use_sys)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/astroid/modutils.py", line 230, in load_module_from_modpath
    module = imp.load_module(curname, mp_file, mp_filename, mp_desc)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/imp.py", line 245, in load_module
    return load_package(name, filename)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/imp.py", line 217, in load_package
    return _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint_django/__init__.py", line 7, in <module>
    from pylint_django import plugin
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint_django/plugin.py", line 9, in <module>
    from pylint_django import transforms  # noqa, pylint: disable=unused-import
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint_django/transforms/__init__.py", line 12, in <module>
    foreignkey.add_transform(MANAGER)
  File "/home/ferd/.virtualenvs/python37/lib/python3.7/site-packages/pylint_django/transforms/foreignkey.py", line 59, in add_transform
    manager.register_transform(nodes.CallFunc, inference_tip(infer_key_classes),
AttributeError: module 'astroid.nodes' has no attribute 'CallFunc'
```
Freeze:
```
astroid==2.0.0.dev4
isort==4.3.4
lazy-object-proxy==1.3.1
mccabe==0.6.1
Pillow==5.2.0
pylint==2.0.0.dev2
pylint-django==0.11.1
pylint-plugin-utils==0.3
six==1.11.0
wrapt==1.10.11
```

**Comment by atodorov** [comment]
@carlio I think astroid has changed some of the node names. I've seen some of the different node names at a pylint workshop I held recently. 

How do we want to handle this? Try to be backwards and forwards compatible or just require newer astroid/pylint as dependencies ?

**Comment by rdrey** [comment]
I'm struggling to install `pylint_django` in py3.7 with `pipenv` because the constraint `astroid<2.0` is not playing well with `pylint>=2.0.0.dev2`. Could you maybe release your own `dev / --pre` version of `pylint-django`?

**Comment by federicobond** [comment]
astroid 2.0 is Python 3 only, so I think we should be compatible with previous versions until support for Django 1.11 is dropped.

If we can get a Pylint 2.x build running in CI (with failures ignored for now), I can tackle some of these issues, but I don’t know enough tox to do it.

**Comment by carlio** [comment]
@atodorov I think that keeping compat/backwards support layers is probably not worth it any more unless it's easy. There are lots of changes to django, astroid, pylint and python so it really becomes hard to keep everything going. Certainly when I tried that with this and other libs it made the code worse ans therefore the library I was writing worse, and added lots of additional vectors for bugs... 

Simple things is fine, see this for example https://github.com/PyCQA/pylint-plugin-utils/blob/master/pylint_plugin_utils/__init__.py#L2  so I think a mild 'compat.py' isn't a huge problem for just renaming nodes etc, but I wouldn't put too much emphasis on keeping old versions of libs working with new versions of pylint-django.

If you're using old pylint/old django then use old pylint-django!

**Comment by carlio** [comment]
pylint 2.0 and android 2.0 just got released which might affect this.

**Comment by atodorov** [comment]
@carlio I'm still not quite sure how to reproduce locally. 

Everything was working fine for me with pylint/astroid 2.0 but an older pylint-plugin-utils. 

With pylint-plugin-utils 0.3 and pylint/astroid 2.0 I see the traceback from comment #0.  I also see you've merged a PR that seems related. Maybe we need a new version of pylint-plugin-utils and to bump the minimum required version in setup.py ?

**Comment by carlio** [comment]
@atodorov I can bump the pylint-plugin-utils release, also I can add you to that team/group if you like (as far as I know only pylint-django uses it a lot, pylint-celery isn't used much).

**Comment by carlio** [comment]
@atodorov pylint-plugin-utils 0.4 just released. I didn't pin the pylint version in that lib, that's something that should be chosen in the pylint-django (or other plugin) libs I feel.

**Comment by ochronus** [comment]
Still getting the same error with pipenv, python 3.7.0. Pip freeze output:

`
astroid==2.0
pylint==2.0.0
pylint-django==0.11.1
pylint-plugin-utils==0.4
`

**Comment by w-flo** [comment]
@ochronus AIUI, it's fixed in master, but not released yet. So we need some patience. ;-)

**Comment by atodorov** [comment]
or a pull request to fix the remaining test failures :)

**Comment by ochronus** [comment]
Oh, my bad, since it was closed I thought it's out in the wild :) thx for the heads up!

**Comment by allthatilk** [comment]
Can I ask what the status is currently on this? I've just started getting this issue myself.

**Comment by gothraven** [comment]
Can't wait for this to be fixed, Thanks a lot 👍

**Comment by federicobond** [comment]
Do not wait, do :)

**Comment by gothraven** [comment]
I'm not that advanced, but sure, i'll try that out

**Comment by katsika** [comment]
Workaround before new version is released:

```
pip uninstall pylint-django
pip --no-cache-dir install git+https://github.com/PyCQA/pylint-django.git@4316c3d90f4ac6cbbeddfc8d431f5d4e031d5cf1
```

**Comment by carlio** [comment]
@atodorov this fixed this problem in another lib I had - https://github.com/landscapeio/requirements-detector/commit/5c97a189a0c673596246cbbf8400f28d1ec6f88b

Not sure if that's how you'd like to fix it here but seems simple enough so might be worth thinking about.

**Comment by carlio** [comment]
@dspacejs temporarily, `pylint<2` and `astroid<2` would probably fix your local build. v2 of both broke a few things in prospector and several dependent libraries so using old versions is probably the quickest fix while we catch up.

**Comment by atodorov** [comment]
@carlio we've already taken care of the node name changes introduced by astroid 2.0. Currently a few test cases are failing. From what I can tell the inference works differently and whenever we try to compare against expected class names (e.g. subclasses of Factory) that fails. Why and how I still can't figure out.

**Comment by dspacejs** [comment]
Installing an older version of `pylint` worked for me as a workaround:
```
pipenv install -d pylint==1.9.3
```

**Comment by w-flo** [comment]
The latest released version of pylint-django (2.0, it's only a few hours old) works for me on python 3.7.
